pandas/_libs/tslibs/period.pyx

```diff
@@ -1155,46 +1155,81 @@ cdef int64_t period_ordinal_to_dt64(int64_t ordinal, int freq) except? -1:
 
 
 cdef str period_format(int64_t value, int freq, object fmt=None):
+
     cdef:
-        int freq_group
+        int freq_group, quarter
+        npy_datetimestruct dts
+        bint is_fmt_none
 
     if value == NPY_NAT:
         return "NaT"
 
-    if isinstance(fmt, str):
-        # Encode using current locale, in case fmt contains non-utf8 chars
-        fmt = <bytes>util.string_encode_locale(fmt)
-
-    if fmt is None:
-        freq_group = get_freq_group(freq)
-        if freq_group == FR_ANN:
-            fmt = b"%Y"
-        elif freq_group == FR_QTR:
-            fmt = b"%FQ%q"
-        elif freq_group == FR_MTH:
-            fmt = b"%Y-%m"
-        elif freq_group == FR_WK:
-            left = period_asfreq(value, freq, FR_DAY, 0)
-            right = period_asfreq(value, freq, FR_DAY, 1)
-            return f"{period_format(left, FR_DAY)}/{period_format(right, FR_DAY)}"
-        elif freq_group == FR_BUS or freq_group == FR_DAY:
-            fmt = b"%Y-%m-%d"
-        elif freq_group == FR_HR:
-            fmt = b"%Y-%m-%d %H:00"
-        elif freq_group == FR_MIN:
-            fmt = b"%Y-%m-%d %H:%M"
-        elif freq_group == FR_SEC:
-            fmt = b"%Y-%m-%d %H:%M:%S"
-        elif freq_group == FR_MS:
-            fmt = b"%Y-%m-%d %H:%M:%S.%l"
-        elif freq_group == FR_US:
-            fmt = b"%Y-%m-%d %H:%M:%S.%u"
-        elif freq_group == FR_NS:
-            fmt = b"%Y-%m-%d %H:%M:%S.%n"
-        else:
-            raise ValueError(f"Unknown freq: {freq}")
+    # fill dts and freq group
+    get_date_info(value, freq, &dts)
+    freq_group = get_freq_group(freq)
+
+    # use the appropriate default format depending on frequency group
+    is_fmt_none = fmt is None
+    if freq_group == FR_ANN and (is_fmt_none or fmt == "%Y"):
+        return f"{dts.year}"
+
+    elif freq_group == FR_QTR and (is_fmt_none or fmt == "%FQ%q"):
+        # get quarter and modify dts.year to be the fiscal year (?)
```

Comment:
the changes themselves look good, I'm just confused by this `(?)`. are you asking for confirmation?

Reply:
yes, "fiscal year" is the wording to confirm in this comment, I wasn't sure